### DISCUSSION TITLE
Connection-blocked blocks everything, not just publishes

### DISCUSSION THREAD

**Discussion by carlhoerberg** [rabbitmq/rabbitmq-server#3384] [Ideas]
From https://www.rabbitmq.com/amqp-0-9-1-reference.html#connection.blocked

> This method indicates that a connection has been blocked and does not accept new publishes. 

But RabbitMQ seems to stop reading completely from the socket after `connection.blocked` is sent. This prevents the client from `ack`:ing, `purge` etc, which could help in a low memory situation.

**Comment by lhoguin** [comment]
What should RabbitMQ do to incoming publishes? Reject them as they come in and drop the body?

**Reply by carlhoerberg** [reply]
No, but Return them? Or accept them, if the client has advertised in the client properties that it supports "connection.blocked", because then the only messages the server would see are what's already been sent but buffered.
